src/structures/typedefs.h

```diff
@@ -30,12 +30,12 @@ namespace vroom {
 // To easily differentiate variable types.
 using Id = uint64_t;
 using Index = uint16_t;
-using Cost = uint32_t;
+using Cost = uint64_t;
 using Gain = int64_t;
-using Distance = uint32_t;
-using Duration = uint32_t;
+using Distance = uint64_t;
+using Duration = uint64_t;
 using Coordinate = double;
-using Capacity = int64_t;
+using Capacity = uint64_t;
```

Comment:
Thanks for the PR.
This change seems dangerous at first glance. Amounts are basically `std::vector<Capacity>` and they are subtracted occasionally (e.g. https://github.com/VROOM-Project/vroom/blob/c238bf6d0590ea384bcc9a3aa84ca9f4fdde1099/src/structures/vroom/raw_route.cpp#L203). It may very well be that all subtractions guarantee that the result is positive, but this change should be tested nonetheless